assets/js/phoenix_live_view.js

```diff
@@ -1274,6 +1274,57 @@ export let DOM = {
   }
 }
 
+class DOMAppendPrependUpdate {
+  constructor(fromEl, toEl, updateType) {
+    let idsAfter = Array.from(toEl.children).map(child => child.id)
+    let idsBefore = []
+
+    let modifiedIds = []
+
+    Array.from(fromEl.children).forEach(child => {
+      idsBefore.push(child.id)
+      if (idsAfter.indexOf(child.id) >= 0) {
+        modifiedIds.push([child.id, child.previousElementSibling && child.previousElementSibling.id])
+      }
+    })
+
+    this.containerID = toEl.id
+    this.updateType = updateType
+    this.modifiedIds = modifiedIds
+    this.newIds = idsAfter.filter(id => idsBefore.indexOf(id) < 0)
+  }
+
+  // We do the following to optimize append/prepend operations:
+  //   1) Track ids of modified elements & of new elements
+  //   2) Any modified elements are put back in the correct position in the DOM tree
+  //      by storing the id of their previous sibling
+  //   3) New elements are going to be put in the right place by morphdom during append. 
+  //      For prepend, we move them to the first position in the container
+  perform() {
+    let el = DOM.byId(this.containerID)
+    this.modifiedIds.forEach(([id, siblingId]) => {
+      if (siblingId) {
+        maybe(document.getElementById(siblingId), sibling => {
+          maybe(document.getElementById(id), child => {
+            sibling.insertAdjacentElement("afterend", child)
```

Comment:
Should we skip this operation if `child.previousElementSibling && child.previousElementSibling.id == subling.id`? Or does the browser already make it a noop?